File: back/route/src/main/java/com/easter/route/domain/record/entity/dto/RecordDto.java
package com.easter.route.domain.record.entity.dto;

import java.time.LocalDateTime;

import org.springframework.data.mongodb.core.geo.GeoJsonLineString;

import com.easter.route.domain.record.entity.Record;
import com.easter.route.domain.record.entity.enums.RecordType;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class RecordDto {
	private String id;
	private String raceId;
	private RecordType recordType;
	private boolean completed;
	private String runningTime;
	private String averagePace;
	private int heartRate;
	private GeoJsonLineString recordedTrack;
	private String routeId;
	private LocalDateTime createdAt;

	public static RecordDto of(Record record) {
		return RecordDto.builder()
			.id(record.getId())
			.recordType(record.getRecordType())
			.completed(record.isCompleted())
			.runningTime(record.getRunningTime())
			.averagePace(record.getAveragePace())
			.heartRate(record.getHeartRate())
			.recordedTrack(record.getRecordedTrack())
			.routeId(record.getRouteId())
			.createdAt(record.getCreatedAt())
			.build();
	}
}

File: Watch/app/src/main/java/com/easter/watch/presentation/dataModel/RecordDto.kt
package com.easter.watch.presentation.dataModel

import android.os.Parcelable
import androidx.versionedparcelable.VersionedParcelize
import kotlinx.android.parcel.Parcelize
import java.time.LocalDateTime

@Parcelize
data class RecordDto(
    val id: String,
    val routeId: String,
    val paceList: List<String>,
    val distanceList: List<Double>,
    val startPoint: String,
    //val recordedTrack: GeoJsonLineString,
    val runningTime: String,
    val averagePace: String,
    val averageHeartRate: Int,
    val distance: Double,
    val createdAt: String
) : Parcelable
